flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/QueryOperationCatalogView.java

```diff
@@ -32,9 +34,26 @@
 public final class QueryOperationCatalogView implements CatalogView {
 
     private final QueryOperation queryOperation;
+    private final @Nullable String originalQuery;
```

Comment:
shall we just reference the `CatalogView origin` here? Less fields and an easier constructor.

Reply:
Makes total sense!